tools/pythonpkg/tests/arrow/test_progress.py

```diff
@@ -0,0 +1,41 @@
+import duckdb
+import os
+import sys
+try:
+    import pyarrow
+    import pyarrow.parquet
+    import numpy as np
+    can_run = True
+except:
+    can_run = False
+
+class TestProgressBarArrow(object):
+
+    def test_progress_arrow(self, duckdb_cursor):
+        if not can_run:
+            return
+
+        data = (pyarrow.array(np.arange(10000000), type=pyarrow.int32()))
+        duckdb_conn = duckdb.connect()
+        duckdb_conn.execute("PRAGMA set_progress_bar_time=1")
+        duckdb_conn.execute("PRAGMA disable_print_progress_bar")
+
+        tbl = pyarrow.Table.from_arrays([data],['a'])
+        rel = duckdb_conn.from_arrow_table(tbl)
```

Comment:
Could you add some tests for the decimals, at least one for each of the different physical types (int16, int32, int64, int128), and also trying to trigger the failure case in Hugeint::TryCast?

Reply:
Not related to the test_progress itself right.

Reply:
Indeed, just a general request for tests

Reply:
I don't think it is possible to trigger a failure in Hugeint::TryCast (at least shouldn't). I guess you are just suggesting to create decimals on precisions that will be transformed to int16,32 and 64, and be sure that the 128 conversions won't explode there.
Is that it or did I miss something?

Reply:
Yes exactly. If it should not be possible to trigger such a failure then that's fine as well, as long as there is an assertion there that at least checks the return value of the TryCast.